### PR TITLE
Print Write Size

### PR DESCRIPTION
**Pull request by Maithem**
Print the write size on an OverwriteException.

### CODE REVIEW

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **d2d2e4f**.

*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 10 threads, Disk mode
 
 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #923 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/S0Y73aIZ4fvdw4tq7UKSS3EPl8IpMOgw)

File: runtime/src/main/java/org/corfudb/runtime/view/StreamsView.java [hunk 1]
```diff
@@ -142,9 +145,17 @@ public long append(@Nonnull Set<UUID> streamIDs, @Nonnull Object object,
                 // If we're here, we succeeded, return the acquired token
                 return tokenResponse.getTokenValue();
             } catch (OverwriteException oe) {
+                int sizeOfWrite = 0;
+                try {
+                    ILogData.SerializationHandle handle = new LogData(DataType.DATA, object).getSerializedForm();
+                    sizeOfWrite = handle.getSerialized().getSizeEstimate();
+                    handle.close();
+                } catch (Exception e) {
+                    log.warn("Couldnt get size", e);
```
Comment (no2chem) [inline runtime/src/main/java/org/corfudb/runtime/view/StreamsView.java]: Log message format not correct

File: runtime/src/main/java/org/corfudb/runtime/view/StreamsView.java [hunk 2]
```diff
@@ -142,9 +145,17 @@ public long append(@Nonnull Set<UUID> streamIDs, @Nonnull Object object,
...
+                }
                 // We were overwritten, get a new token and try again.
-                log.warn("append[{}]: Overwrite , streams {}", tokenResponse.getTokenValue(),
-                        streamIDs.stream().map(Utils::toReadableId).collect(Collectors.toSet()));
+                log.warn("append[{}]: Overwrite , streams {}, size {}", tokenResponse.getTokenValue(),
```
Comment (no2chem) [inline runtime/src/main/java/org/corfudb/runtime/view/StreamsView.java]: Does this really need to be warn level?

File: runtime/src/main/java/org/corfudb/runtime/view/StreamsView.java [hunk 3]
```diff
@@ -142,9 +145,17 @@ public long append(@Nonnull Set<UUID> streamIDs, @Nonnull Object object,
                 // If we're here, we succeeded, return the acquired token
                 return tokenResponse.getTokenValue();
             } catch (OverwriteException oe) {
+                int sizeOfWrite = 0;
```
Comment (no2chem) [inline runtime/src/main/java/org/corfudb/runtime/view/StreamsView.java]: This calculation should only be done if the log is set to output warn level messages

File: runtime/src/main/java/org/corfudb/runtime/view/StreamsView.java [hunk 4]
```diff
@@ -142,9 +145,17 @@ public long append(@Nonnull Set<UUID> streamIDs, @Nonnull Object object,
...
+                try {
+                    ILogData.SerializationHandle handle = new LogData(DataType.DATA, object).getSerializedForm();
```
Comment (no2chem) [inline runtime/src/main/java/org/corfudb/runtime/view/StreamsView.java]: Probably should be a static method inside of ILogData.

**Comment by Maithem** [comment]
addressed by https://github.com/CorfuDB/CorfuDB/pull/925

**Pull request closed without merging**